modules/tinymce/src/core/main/ts/api/EditorUpload.ts

```diff
@@ -98,6 +98,15 @@ interface EditorUpload {
   destroy: () => void;
 }
 
+const isEmptyForPadding = (element: SugarElement<any>) =>
+  !Traverse.hasChildNodes(element) && SugarNode.isTag('p')(element);
```

Comment:
There is more cases than `P` here basically any text block element. Things like TD or LI will be hard since they are part of a bigger structure. So you need the schema here and use schema.getTextBlocks() then check if the SugarNode.name is in that map. Then this will work for H1, DIV etc etc. Add tests for a few other types as well.

- [ ] Expand to more than just paragraphs